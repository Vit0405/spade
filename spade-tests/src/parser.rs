use crate::snapshot_error;

snapshot_error! {
    stage_outside_pipeline,
    "
    entity main(x: X) -> int<8> {
        reg;
    }
    "
}

snapshot_error! {
    register_count_is_required,
    "
    pipeline(3) main(x: X) -> int<8> {
        reg *;
    }
    "
}
