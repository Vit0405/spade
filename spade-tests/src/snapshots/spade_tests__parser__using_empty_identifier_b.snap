---
source: spade-tests/src/parser.rs
---
use conv;

entity counter(clk: clock, rst: bool, max: int<8>) -> int<8> {
    reg(clk) value reset (rst: 0) =
        if value == max {
            0
        }
        else {
            conv::trunc(value + 1)
        };
    value
}


error: Use of undeclared name conv::trunc
  ┌─ testinput:9:13
  │
9 │             conv::trunc(value + 1)
  │             ^^^^^^^^^^^ Undeclared name


